src/components/useFetch.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState, useEffect } from 'react'
+
+const useFetch = (url: string) => {
+    type post = {
```

Comment:
You should define interfaces and types out of the function declaration so it's not created on each function call.

Reply:
Fixed.